content/blog/testthat-3-1/index.Rmd

```diff
@@ -0,0 +1,157 @@
+* [x] Create `thumbnail-sq.jpg`; height and width should be equal
+* [x] Create `thumbnail-wd.jpg`; width should be >5x height
+* [x] `hugodown::use_tidy_thumbnails()`
+* [x] Add intro sentence, e.g. the standard tagline for the package
+* [x] `usethis::use_tidy_thanks()`
+-->
+
+We're stoked to announce the release of [testthat](http://testthat.r-lib.org/) 3.1.0.
+testthat makes it easy to turn your existing informal tests into formal, automated tests that you can rerun quickly and easily.
+testthat is the most popular unit-testing package for R, and is used by over 6,000 CRAN and Bioconductor packages.
+You can learn more about unit testing at <https://r-pkgs.org/tests.html>.
+
+You can install testthat from CRAN with:
+
+```{r, eval = FALSE}
+install.packages("{package}")
+```
+
+This release of testthat includes a bunch of minor improvements to snapshotting as well as one breaking change (which only applies if you're using the 3rd edition).
+You can see a full list of changes in the [release notes](https://github.com/r-lib/testthat/blob/master/NEWS.md).
+
+```{r setup}
+library(testthat)
+```
+
+## Snapshot tests
+
+Most of the effort in this release has gone in [snapshot tests](https://testthat.r-lib.org/articles/snapshotting.html), a new feature in testthat 3.0.0.
+snapshot tests (also known as [golden tests](https://ro-che.info/articles/2017-12-04-golden-tests)) record expect output in a separate human-readable file Instead of using code to describe expected output.
+Since the release of testthat 3.0.0, we've started using snapshot tests across a bunch of tidyverse packages and they've been working out really well.
+I don't anticipate any major changes (although we may continue to add new features) so the snapshot functions have changed lifecycle stages from experimental to **stable**.
+
+This release also includes two new features that help you use snapshot tests in more places:
+
+-   `expect_snapshot()` gains a `transform` argument, which should be a function that takes a character vector of lines and returns a modified character vector of lines.
+    This makes it easy to remove sensitive (e.g. API keys) or stochastic (e.g. random temporary directory names) data from snapshot output.
+
+    ```{r}
+    get_info <- function() {
+      list(
+        name = "Hadley", 
+        password = "sssh-its-a-secret"
+      )
+    }
+
+    hide_password <- function(x) {
+      is_password <- grepl("password", x)
+      ifelse(is_password, "<REDACTED>", x)
+    }
+
+    test_that("info retruns name and password", {
+      local_edition(3)
+      expect_snapshot(str(get_info()), transform = hide_password)
+    })
+    ```
+
+    If you need `transform`, I recommend designing your printing methods so the output can be easily manipulated with regexps.
+
+-   `expect_snapshot()` and friends gets an experimental new `variant` argument which causes the snapshot to be saved in `_snaps/{variant}/{test}.md` instead of `_snaps/{test}.md`.
```

Comment:
```suggestion
-   `expect_snapshot()` and friends get an experimental new `variant` argument which causes the snapshot to be saved in `_snaps/{variant}/{test}.md` instead of `_snaps/{test}.md`.
```